packages/medusa/src/services/order.ts

```diff
@@ -298,20 +317,22 @@ class OrderService extends BaseService {
 
   /**
    * Gets an order by id.
-   * @param {string} orderId - id of order to retrieve
-   * @param {Object} config - config of order to retrieve
-   * @return {Promise<Order>} the order document
+   * @param orderId - id of order to retrieve
+   * @param config - config of order to retrieve
+   * @return the order document
    */
-  async retrieve(orderId, config = {}) {
+  async retrieve(
+    orderId: string,
+    config: FindConfig<Order> = {}
+  ): Promise<Order> {
     const orderRepo = this.manager_.getCustomRepository(this.orderRepository_)
-    const validatedId = this.validateId_(orderId)
 
     const { select, relations, totalsToSelect } =
       this.transformQueryForTotals_(config)
 
     const query = {
-      where: { id: validatedId },
-    }
+      where: { id: orderId },
+    } as FindConfig<Order>
```

Comment:
question(non-blocking): is there a difference between

```
    const query: FindConfig<Order> = {
      where: { id: orderId },
    }

and

    const query = {
      where: { id: orderId },
    } as FindConfig<Order>
```

Have seen both used but not sure if there is a significant difference?

Reply:
There is a some differences :
1. variable: FindConfig means that the variables must be of that type and therefore require all the non optional properties to be set even with a default balue that you wouldn’t want to set because you will override it just after.
2. Cast using as FindConfig means that we are forcing the type because we are assigning a sub set of the expected props, bu we know for example that we will assign the other props later on and that we want to tell to the compiler that a props exists on it, can be useful when for example on the query you want to assign a props later depending on some logic but that you dont want to initialise it here with a default value that will be overridden later on but you still want the compiler to recognise that props as part of the variable. It can be used in other cases as well.

let me know if i did not explain it properly 😅